src/simple_http.rs

```diff
@@ -177,24 +177,36 @@ impl SimpleHttpTransport {
 
         // Send HTTP request
         {
-            let mut sock = BufWriter::new(sock.get_mut());
-            sock.write_all(b"POST ")?;
-            sock.write_all(self.path.as_bytes())?;
-            sock.write_all(b" HTTP/1.1\r\n")?;
+            let mut write_sock = BufWriter::new(sock.get_mut());
+            // When we write to a socket, it may have died but we do not detect it. In this case we
+            // want to detect this ASAP and reconnect. We do this by writing the literal text POST
+            // in two pieces and checking for error returns on either one, and retrying in this
+            // case.
+            //
+            // From http://www.softlab.ntua.gr/facilities/documentation/unix/unix-socket-faq/unix-socket-faq-2.html
+            // "If the peer calls close() or exits...I would expect EPIPE, not on the next call,
+            // but the one after."
+            if write_sock.write_all(b"PO").is_err() || write_sock.write_all(b"ST ").is_err() {
+                **write_sock.get_mut() = self.fresh_socket()?;
+                write_sock.write_all(b"POST ")?;
+            }
+            write_sock.write_all(b"POST ")?;
```

Comment:
Why the second write of "POST" (excluding the error path)? At this stage we've either written POST already or returned an error?

Reply:
Oh, derp, this second POST should not be there and probably completely breaks things. Pushed to remove it.